test/e2e/utils/recordedit-helpers.js

```diff
@@ -193,7 +193,26 @@ exports.testPresentationAndBasicValidation = function(tableParams, isEditMode) {
             });
         });
     });
-
+    
+    if (tableParams.table_name === "accommodation") {
+        it("should give a warning when leaving the edit page with unsaved changes", function (done) {
+            var EC = protractor.ExpectedConditions;
+            var ratingInput = element(by.name("rating"));
+            browser.wait(EC.visibilityOf(ratingInput, browser.params.defaultTimeout));
+            ratingInput.sendKeys("5");
+            browser.refresh();
+            browser.switchTo().alert().then(function (alert) {
```

Comment:
Only thing missing from this test case is an `expect()` clause. Not sure what will happen without the expectation.. 🤔 but to the best of my knowledge, the test will pass without actually running (unless done changes that fact).

I think we want to do something like:
```js
expect(alert.getText()).toBe("Do you want to leave this page? Changes you have made will not be saved.", "Alert message is not what we expected");
```

Reply:
So the alert behaves differently on different browsers. But on most browsers, it doesn't return the custom text that I have passed while calling the alert. I think alert().accept() is sufficient in this case because if the alert doesn't pop up at that point in the test case we'll get an error saying: "Failed: no such alert", i.e. we are trying to accept an alert that is not even present on the screen.